grails-app/controllers/org/broadinstitute/orsp/api/DataUseController.groovy

```diff
@@ -89,6 +93,18 @@ class DataUseController extends AuthenticatedController {
         }
     }
 
+    def saveSdul() {
+        try {
+            DataUseRestriction restriction = dataUseLetterService.updateRestrictionFromParams(request.JSON)
```

Comment:
If consent group doesn't have a restriction it will be created in DataUseRestrictionParser. This is why I am not returning 404. If exists it uses the existent one. Does this clarify the logic?
dataUseRestriction = dataUseRestriction == null ? new DataUseRestriction() : dataUseRestriction

Reply:
For the 404 question, yes, this makes sense that you don't need to handle not-found. The 400 handling is still not correct though, I'll comment again in the updated code.